lib/traject/indexer.rb

```diff
@@ -568,15 +568,19 @@ def inspect
 
     def execute(context)
       accumulator = []
-      [@lambda, @block].each do |aProc|
-        next unless aProc
-
-        if aProc.arity == 2
-          aProc.call(context.source_record, accumulator)
+      if @lambda
+        if @lambda.arity == 2
+          @lambda.call(context.source_record, accumulator)
         else
-          aProc.call(context.source_record, accumulator, context)
+          @lambda.call(context.source_record, accumulator, context)
+        end
+      end
+      if @block
+        if @block.arity == 2
+          @block.call(context.source_record, accumulator)
+        else
+          @block.call(context.source_record, accumulator, context)
         end
```

Comment:
Ah wait, it makes no sense to put `[@lambda, @block]` into a global constant, since those are iVars of course. 

Hmm. Any other patterns we could use to avoid the array creation while keeping it DRY? A submethod instead? Something like:

    call_proc(@lambda)
    call_proc(@block)

Method call isn't prohibitively expensive, is it?  Sorry to be picky here, but I think us being picky and careful so far is part of what's led to traject being what it is. 

Do you have any scripts or tests you use for benchmarking/profiling you can share, so others like me or Bill can try alternatives and see if they perform just as well with cleaner code? (Might apply to future too, when someone looks to alter or refactor some of this code, and wants to make sure they arent' performance regressing your changes).